### PR TITLE
Fix dependency lookup

### PR DESCRIPTION
**Pull request by mydea**
This PR changes some things to make the CLI commands work with Yarn Workspaces.

Previously, we did a lot of absolute path lookup, like `node_modules/xgettext-template/...`. This used to work fine, but breaks in scenarios like Yarn Workspaces, where dependencies might be somewhere else.

Instead, we now use the node dependency resolution (e.g. `require.resolve('xgettext-template')`) for this. 

This also means that we don't need to include the xgettext-template & gettext-parser dependencies in the host app - you can safely remove it from your own app's package.json, as it will now be loaded via ember-l10n's dependencies.

In addition to that, this PR also made some improvements for Windows:

1. It fixes the dependency check to work on Windows. In the process, I also removed the npm dependency checks, as these dependencies are now included anyhow.
2. It fixes the line number comments for .hbs files, which previously had wrong slashes (e.g. `./app\components\...`).

Finally, this PR also removes the temporary .hbs file that we generated without whitespace. The current version of xgettext-template seems to be able to handle whitespace just fine. This makes the line comments more meaningful. (If this used to only fail under very specific circumstances, please let me know, then I can remove this change again).

I tried all these changes on Windows, in two different apps (one using Workspaces), and everything seemed to work fine. But we should definitely also verify if everything (e.g. the dependency check, extraction, syncing, ...) also continues to work fine on Mac OS.

PS: This PR also updates all dependencies to their latest version & moves the tests to the new test syntax.

### CODE REVIEW

File: lib/commands/extract.js [hunk 1]
```diff
@@ -502,6 +502,16 @@ EXTRACTING TEMPLATE TRANSLATIONS
...
+      // Fix paths for Windows - there, xgettext-template makes paths like ./app\components\...
+      let fileContent = fs.readFileSync(tmpPOTFile, encoding);
+
+      let regex = /#: (.*)(\\)(.*)\.hbs/gm;
+      while (regex.exec(fileContent)) {
+        fileContent = fileContent.replace(regex, '#: $1/$3.hbs');
+      }
+
+     fs.writeFileSync(tmpPOTFile, fileContent);
+
```
Comment (arm1n) [inline lib/commands/extract.js]: I'm not happy with that change, it introduces another file I/O and the HBS extraction is already very slow. The comments are for information only and window paths are backslashs. So I would suggest to normalize the paths in the extraction methods, so you wouldn't get a mixture. Hope that's fine - please also check this again on your machine.

**Comment by arm1n** [comment]
All other changes look good and are fine with me, thanks for your efforts. Please check my two commits and check again if it's working on windows too. On unix systems is working fine now. After your final check I will merge it and publish a new version.

**Comment by mydea** [comment]
I added two more commits, one fixing the eslint issues, and one using a forked version of xgettext-template until this https://github.com/gmarty/xgettext/pull/73 is hopefully merged. There, I made the changes to fix the Windows paths upstream. (I wouldn't care too much about the paths if it wouldn't result in a lot of Git noise when the comments are changed every time someone extracts from a different OS).

I tried this on Windows and everything seemed to work well for me!

**Comment by arm1n** [comment]
This is fine, but as I've seen the discussion in the pull request gmarty/xgettext#73 we should wait until merged and publish our new version then, so we can save from publishing overhead and using a fork. Hope you are fine with that, there's no pressure or immediate bugfix to get out.

**Comment by mydea** [comment]
Sure, that's fine. I guess the only "pressing" need is to use this in a Yarn workspace app, but we can safely use this Git branch for that until we feel comfortable releasing an actual version.

**Comment by mydea** [comment]
That was quick, it's already released as 3.4.2, I updated the PR accordingly. From my point of view, this would be ready to merge & release.

Not sure if we maybe should release this as 1.2? It's not breaking, but still changes a bit (e.g. you can remove the manual xgettext-template & gettext-parser dependencies, ...). What do you think @arm1n ?

**Comment by arm1n** [comment]
Nice, that was really quick :) Yeah I agree, we should release it as 2.1 and add notes to the release.

**Pull request merged**